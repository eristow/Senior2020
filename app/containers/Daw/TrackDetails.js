import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Container = styled.div`
  width: 100%;
  height: auto;
  background: lightgray;
  margin-top: 10px;
  border: 2px solid black;
  border-radius: 5px;
`;

const Upper = styled.p`
  text-transform: uppercase;
`;

export function TrackDetails({ track }) {
  return (
    <Container>
      {track === null ? (
        <Upper>Please select a track to modify.</Upper>
      ) : (
        <Upper>{track}</Upper>
      )}
    </Container>
  );
}

TrackDetails.propTypes = {
  track: PropTypes.string,
};

export default TrackDetails;
