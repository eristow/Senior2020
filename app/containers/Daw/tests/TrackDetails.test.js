import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';

import { TrackDetails } from '../TrackDetails';

describe('<TrackDetails />', () => {
  let store;
  const track = { name: 'Track 1' };

  beforeAll(() => {
    store = configureStore();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();

    render(
      <Provider store={store}>
        <TrackDetails dispatch={dispatch} track={track} />
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <TrackDetails track={track} />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
