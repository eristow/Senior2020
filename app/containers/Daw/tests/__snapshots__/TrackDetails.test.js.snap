// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackDetails /> Should render and match the snapshot 1`] = `
<div
  class="TrackDetails__Container-sc-1pg6vl2-0 ImYeq"
>
  <p
    class="TrackDetails__Upper-sc-1pg6vl2-1 cgsodF"
  >
    Track 1
  </p>
</div>
`;
